### DISCUSSION TITLE
Using USB OTG HS with internal FS PHY on STM32F446RET6

### DISCUSSION THREAD

**Discussion by peppapighs** [hathach/tinyusb#2867] [Q&A] [answered]
Hi,

I am currently working on a simple DFU bootloader on my STM32F446RET6 PCB. I have my USB port connected to the USB OTG HS pins and I plan to use the internal full speed PHY as described in datasheet. Here is my configuration:

![image](https://github.com/user-attachments/assets/141cd8ab-3e63-430f-b879-89d1fec1c0c3)

I have added to following definitions in the `tusb_config.h`:

```c
#define CFG_TUSB_MCU OPT_MCU_STM32F4

#define BOARD_DEVICE_RHPORT_SPEED OPT_MODE_FULL_SPEED

#define BOARD_DEVICE_RHPORT_NUM 1
#define CFG_TUSB_RHPORT1_MODE (OPT_MODE_DEVICE | OPT_MODE_FULL_SPEED)
#define BOARD_TUD_RHPORT 1
```

However, when I flash and run the firmware, the following assertion always fails:

https://github.com/hathach/tinyusb/blob/038be21e4d94c958043eb54d179948cd9e52fe18/src/portable/synopsys/dwc2/dcd_dwc2.c#L647

This does not happen when I am using my development board with the USB port connected to USB OTG FS. Any idea what the problem is here?

Thank you.

**Comment by HiFiPhile** [comment]
Hi, what's the value of gsnpsid ? If it's 0 then maybe OTG_HS's clock is not enabled.

**Reply by peppapighs** [reply, marked as answer]
Hi, it seems that `gsnpsid` is 0 as you suggested. Investigating the debugger, I found out at the `rhport` is 0 so it turns out that running `tusb_init()` without supplying any arguments will default to RH port 0 so I have to call `tusb_init(1, NULL)` instead. The issue should be fixed now. Thank you very much!
